transcriptic/commands.py

```diff
@@ -913,7 +922,8 @@ def login(api, config, api_root=None, analytics=True):
                 '200': lambda resp: resp,
                 '401': lambda resp: resp,
```

Comment:
This needs to be fixed in the webapp, so that one cant get a token without properly signing the request.

Reply:
It should fail there rather than later on

Reply:
Yea, agreed. It would make much more sense to fail here rather than later.

Thinking about the flow in the near future if we were to incorporate uploading the key via the CLI... it seems logical to make key uploading part of the login flow, but if we were to also demand having the key in place before being able to get a token, then the flow will look like
1. Upload the key (minimally requires User and key path)
2. Authenticate token (requires User, token and key path)

Some open questions there would be:
1. Do we always upload the key? Or should it be gated by a prompt/option
2. Do we separate the key/password prompts? I.e. User + Key first, and only if that's valid, pw?

Reply:
Hmm... looking at the route for uploading the key, sounds like we require a valid token? This seems like a chicken and egg problem then, we probably do not actually want to require the key for token validation.

So the flow would probably look like
1. Authenticate token (requiring just User and token)
2. Uploading the key (requiring user, token and key)

Hmm. Maybe we do want to add an additional step between this and the next select_organization call then to validate the token? Or at least catch it with a clearer error message if we're relying on the subsequent select_organization call to do the validation

Reply:
Yeah, we wouldnt want, for example, to let User A upload a public key for User B, so we need to require the token

Reply:
To me, the right course of action is the following:

1. User runs cli like `transcriptic login [--enable-signing path/to/key.pem]`
2. User is prompted for Username/Pass
3. User/Pass[/Key] used to acquire token
4. If #3 fails, we probably don't want to expose which credential (Pass/Key) was incorrect to be safe, so we should likely fail outright
5-∞ assuming #3 succeeds, we now have all proper crednetials

Reply:
I'm not sure the most robust way of going about this, I agree it should be an 401 coming back from the server. However, 401  wont always mean a missing signature, so we'd have to do some text parsing, which I don't  like so much. Maybe it is better to change the message to be more informative on the ruby side?

Reply:
@yangchoo